### DISCUSSION TITLE
How to get the parent class of widget when on click is fired

### DISCUSSION THREAD

**Discussion by MohamedElgamal** [Textualize/textual#3198] [Q&A] [answered]
Hello everyone , having the following scenario i have class called StateItem that contain a collection of widget and checkbox and the StateItem class is called in class called StateTabContent() that create more than one object of StateItem depend the data saved in database, the checkbox that life in StateItem object i want when i clicking on it send the data member of StateItem to database
this is my following code : 
<pre>
class StateItem(Static):
    def __init__(self, username, state, request):
        super().__init__()
        self.current_logged_user = username
        self.state = state
        self.request = request

    def compose(self) -> ComposeResult:
        yield Horizontal(
            Label(id="issue-type", renderable=self.request.get_type())
            , Label(id="issue-user", renderable=self.request.get_user())
            , Label(id="issue-details", renderable=self.request.get_request_details())
            , Label(id=f"issue-created-at-{self.request.get_id()}", renderable=str(self.request.get_created_at()))
            , Checkbox(classes="issue-checkbox")
            , classes="state-item"
            , id=f"horizontal-con-{self.request.get_id()}"
        )

    def on_mount(self):
        pass

    @on(Checkbox.Changed, ".issue-checkbox")
    def state_request_checkbox_clicked(self, event):
        is_completed = admin_dao.move_request_to_next_state(self.request.get_id, self.state.get_id())
        self.remove()

class StateTabContent(Widget):
    def __init__(self, username, state):
        super().__init__()
        self.current_logged_user = username
        self.state: StateModel = state

    def compose(self) -> ComposeResult:
        requests_state_records = admin_dao.get_requests_records(self.state.get_id())
        with VerticalScroll():
            for request in requests_state_records:
                yield StateItem(self.current_logged_user, self.state, request)
</pre>

and screen output 

![image](https://github.com/Textualize/textual/assets/25607813/9d513446-364b-47cb-965c-1d56fa8a082b)
the following is happen to me when i click on checkbox nothing happen only the widget is removed ,i know the problem is coming from it can't access the data member of StateItem object but i don't know how to fix it

**Comment by davep** [comment]
I'm struggling to follow the question, or quite see how it relates to the title of the discussion, but to answer the question in the title: a widget's parent is available by its `parent` property.

As for the main body of the question above, do you think you could try and make a small stand-alone example of what it is you want to do and explain what's not working?

**Comment by MohamedElgamal** [comment]
what i want when the checkbox is clicked to access the member attribute of the class who composed the checkbox to sent it to database
sample code of what i tried to achieve 
<pre>
class StateItem(Static):
    def __init__(self, value):
        super().__init__()
        self.saved_value = value

    def compose(self) -> ComposeResult:
        yield Horizontal(
            Label(id="issue-type", renderable=self.request.get_type())
            , Checkbox(classes="issue-checkbox")
            , id="horizontal-con"
        )

    def on_mount(self):
        pass

    @on(Checkbox.Changed, ".issue-checkbox")
    def state_request_checkbox_clicked(self, event):
        // i want to get the saved_value attribute of StateItem here
        database.sendvalue(self.saved_value)
        self.remove()
</pre>

**Reply by davep** [reply, marked as answer]
I'm afraid that doesn't really clarify the question. In that example code `state_request_checkbox_clicked` is a method of `StateItem`, so `self.saved_value` would be the `saved_value` attribute of the widget that `composed` the `Checkbox`.

**Comment by MohamedElgamal** [comment]
when i tried self.saved_value the returned value is None so i think the self keyword in state_request_checkbox_clicked is relate to checkbox class not StateIten

**Reply by davep** [reply]
Given the code you've shown above, `saved_value` is a property of on instance of `StateItem`. If you are accessing it and the value is `None` that means it *does* exist as a property of an instance of `StateItem`, but that its value is `None`. Textual's `Checkbox` doesn't have a property called `saved_value`. If you were somehow attempting to access `saved_item` on an instance of a `Checkbox` that would be an `AttributeError`.

**Comment by MohamedElgamal** [comment]
i figured out the problem i forget brackets 
at this line : 
        is_completed = admin_dao.move_request_to_next_state(self.request.get_id, self.state.get_id())
sorry bro i have another question if you don't mind 
 I have three tabs in my application, namely "Ready," "In Progress," and "Completed," the content of each tab is fetched from the database. I want to ensure that when a tab is clicked, the data is fetched again because it might have been changed or updated.

**Reply by holdenweb** [reply]
This kind of question is often more quickly answered in the Discord web site: https://discord.com/channels/1026214085173461072/1033754296224841768

**Reply by TomJGooding** [reply]
Just to confirm, this question above was posted in a new discussion #3201 (which made sense as on a different topic) and answered by davep within the hour. Of course nothing wrong with instead asking for help on Discord!

**Reply by davep** [reply]
Yup, as Tom says, here is just as good and as monitored as Discord.
